python_modules/libraries/dagster-components/dagster_components/components/dbt_project/component.py

```diff
@@ -169,14 +157,16 @@ class DbtProjectComponent(Component, ResolvedFrom[DbtProjectModel]):
     """
 
     dbt: Annotated[DbtCliResource, Resolver(resolve_dbt)]
-    op: Annotated[Optional[OpSpec], Resolver.from_annotation()] = None
-    # This requires from_parent because it access asset_attributes in the model
-    translator: Annotated[DagsterDbtTranslator, Resolver.from_model(resolve_translator)] = field(
-        default_factory=DagsterDbtTranslator
-    )
-    asset_post_processors: Annotated[
-        Optional[Sequence[AssetPostProcessor]], Resolver.from_annotation()
-    ] = None
+    op: Optional[OpSpec] = None
+    translator: Annotated[
+        DagsterDbtTranslator,
+        Resolver.from_model(
+            resolve_translator,
+            model_field_name="asset_attributes",
+            model_field_type=Optional[Union[str, AssetAttributesModel]],
+        ),
```

Comment:
I quite like this, to be honest. Definitely verbose but not overly-so. `model_field_name` and `model_field_type` are very explanatory IMO. 

Had a thought about cases where the user might want to pass in a full pydantic `Field` object instead of just the type annotation, but then realized they can just do that with `Annotated`